frontend/src/SettingsDashboard.jsx

```diff
@@ -14,18 +14,20 @@ const mapDispatchToProps = {
     putSetting,
 }
 
-class NewSettingsModalInner extends React.Component {
+class SettingsModalInner extends React.Component {
     static propTypes = {
         type: PropTypes.oneOf(['persistent', 'transient']).isRequired,
         onClose: PropTypes.func.isRequired,
         putSetting: PropTypes.func.isRequired,
-        open: PropTypes.bool.isRequired,
     }
 
-    state = {
-        setting: '',
-        value: '',
-    };
+    constructor(props) {
+        super(props);
+        this.state = {
+            setting: props.setting || '',
+            value: props.value || '',
```

Comment:
These would need to be added to the propTypes. BTW, React recommend naming them `initialSetting` and `initialValue` to avoid confusion when copying to the state